components/size_aware_image.jsx

```diff
@@ -108,22 +130,76 @@ export default class SizeAwareImage extends React.PureComponent {
         return null;
     }
 
-    renderImageOrPlaceholder = () => {
+    renderImageWithContainerIfNeeded = () => {
         const {
-            dimensions,
             fileInfo,
             src,
             ...props
         } = this.props;
 
-        let placeHolder;
-        let imageStyleChangesOnLoad = {};
+        Reflect.deleteProperty(props, 'showLoader');
+        Reflect.deleteProperty(props, 'onImageLoaded');
+        Reflect.deleteProperty(props, 'onImageLoadFail');
+        Reflect.deleteProperty(props, 'dimensions');
+        Reflect.deleteProperty(props, 'handleSmallImageContainer');
+
         let ariaLabelImage = localizeMessage('file_attachment.thumbnail', 'file thumbnail');
         if (fileInfo) {
             ariaLabelImage += ` ${fileInfo.name}`.toLowerCase();
         }
 
-        if (dimensions && dimensions.width && !this.state.loaded) {
+        const image = (
+            <img
+                {...props}
+                aria-label={ariaLabelImage}
+                tabIndex='0'
+                onKeyDown={this.onEnterKeyDown}
+                className={this.props.className}
+                src={src}
+                onError={this.handleError}
+                onLoad={this.handleLoad}
+                style={this.props.handleSmallImageContainer && this.state.isSmallImage ? {
+                    boxShadow: 'none',
+                    borderWidth: 0,
+                    borderRadius: 0,
+                    width: 'unset',
+                    margin: 0,
+                    maxHeight: 'inherit',
+                } : {}}
+            />
+        );
+
+        if (this.props.handleSmallImageContainer && this.state.isSmallImage) {
+            let className = 'small-image__container cursor--pointer a11y--active';
+            if (this.state.imageWidth < MIN_IMAGE_SIZE) {
+                className += ' small-image__container--min-width';
+            }
+
+            return (
+                <div
+                    onClick={this.props.onClick}
+                    className={className}
+                    style={this.state.imageWidth > MIN_IMAGE_SIZE ? {
```

Comment:
Same here with respect to inline styles

Reply:
This is using the actual image size after it has loaded, not a predictable/fixed value that can be set in a CSS class, and I don't if there could be a practical way to sue something like `calc` to set it in CSS either.

Reply:
`width: fit-content;` works, but according to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/width), it is experimental and shouldn't be used in production code. And here are more details from [can I use](https://caniuse.com/#search=fit-content)

Reply:
Okay, we can leave this one.